pulp_container/app/tasks/sync_stages.py

```diff
@@ -353,9 +353,9 @@ async def handle_blobs(self, manifest_dc, content_data):
             blob_dc = self.create_blob(layer)
             manifest_dc.extra_data["blob_dcs"].append(blob_dc)
             await self.put(blob_dc)
-        layer = content_data.get("config", None)
-        if layer:
-            blob_dc = self.create_blob(layer, deferred_download=False)
+        config = content_data.get("config", None)
+        if config:
+            blob_dc = await self.create_config_blob(config)
```

Comment:
I don't think it matters if it is a config blob or binary blob. You need to adjust `create_blob` that depending on size, the blob will be downloaded and save into DB or a Declarative Artifact will be created and passed further down to the pipeline.
Manifest.json file has `size` for each layer descriptor. It is a mandatory field so you can use it.

Reply:
I forgot about docker schema1, its' json schema does not have the size field. For this corner case ( and i consider it as a corner because nowadays schema1 is quite rarely used, it is a legacy format) and you can just decide to store all the layers on disk in case of size field absence. This schema1 anyway  does not contain config layer